lib/transaction.js

```diff
@@ -1,15 +1,14 @@
 'use strict';
 
-const Bluebird = require('bluebird');
 const { Model } = require('./model/Model');
 const promiseUtils = require('./utils/promiseUtils');
 const { isSubclassOf } = require('./utils/classUtils');
 const { isFunction } = require('./utils/objectUtils');
 
-function transaction() {
+async function transaction() {
   // There must be at least one model class and the callback.
   if (arguments.length < 2) {
-    return Bluebird.reject(
+    return Promise.reject(
```

Comment:
can probably change all these `Promise.reject(new Error(...` to straight throws inside the async methods